.github/workflows/publish.yml

```diff
@@ -77,6 +76,11 @@ jobs:
           shopt -s globstar
           docker run -v "${PWD}:${PWD}" ghcr.io/linz/argo-tasks:v3 stac-validate "$PWD"/stac/**/collection.json
 
+      - name: Save STAC catalog artifact
```

Comment:
Is there any point in generating the catalog here, we already generate it in `main` step then throw it away, all the generation logic could be just moved into the sync-stac step.

Reply:
The reason was to minimise the amount of non-parallelisable steps.

Reply:
The only step that needs to be singular is the deployment of the STAC to the s3 bucket so we dont get partially updated or stale files.

the stac generation logic here is only used for the publishing to s3, so there isn't much point generating it here to then store as a artifact to download later then upload to s3

Reply:
> The only step that needs to be singular is the deployment of the STAC to the s3 bucket so we dont get partially updated or stale files.

Exactly, that's why the `stac-sync` call is basically the only thing in the `sync-stac` job.

> the stac generation logic here is only used for the publishing to s3, so there isn't much point generating it here to then store as a artifact to download later then upload to s3

How else can I split the two into separate jobs?

Reply:
That said, we should remove STAC validation from the `main` job. No point doing it twice.

Reply:
Don't we need to validate the STAC before we trigger the job to copy the item TIFF and STAC files?

Reply:
> That said, we should remove STAC validation from the main job. No point doing it twice.

The STAC validation runs on pull request so it has to be in the `main` job. removing it from the main job means it will not be validated until its being deployed.

Reply:
Why do we do it twice? Can I just revert df4bddb51188238d23295b39bf6872adacfeb131, save it as an artifact, and remove it from `deploy-prod`?

Reply:
It generally takes ~1 second to generate so its faster to just generate it multiple times than try and deal with the annoyance of artifacts.

I have in the past had artifacts taking 10+ seconds to download and then it is sometimes difficult to know if the right artifact was pulled in.

Reply:
If we do it twice then we have a timing issue - the thing we validate in the main job isn't necessarily the same as the thing we validate in the deploy-prod job. But I've reverted df4bddb51188238d23295b39bf6872adacfeb131.